vowpalwabbit/distributionally_robust.cc

```diff
@@ -162,6 +162,34 @@ ScoredDual ChiSquared::recompute_duals()
   return duals;
 }
 
+void ChiSquared::save_load(io_buf& model_file, bool read, bool text, const char* name)
+{
+  if (model_file.num_files() == 0) { return; }
+  std::stringstream msg;
+
+#define save_load_field(field)                                                            \
```

Comment:
I've added equivalent functionality with this PR: #3326. It's a tiny bit more verbose but does not rely on implicit variable names.

```c++
save_load_field(n);
```
becomes
```c++
VW::process_model_field(io, n, read, "_chisq_n", text);
```

Reply:
It won't work OOTB because I need to dynamically create the read string, but it's worth simplifying the macro.

Reply:
What do you mean by that? I believe it is supported. Please see the doc for the func: https://vowpalwabbit.org/docs/vowpal_wabbit/cpp/latest/namespaceVW_1_1model__utils.html#a8369057d5016e6cfdf02f0ea80570572